substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -549,9 +551,18 @@ impl<T: Config> PoolMember<T> {
 
 	/// Total balance of the member, both active and unbonding.
 	/// Doesn't mutate state.
-	#[cfg(any(feature = "try-runtime", feature = "fuzzing", test, debug_assertions))]
-	fn total_balance(&self) -> BalanceOf<T> {
-		let pool = BondedPool::<T>::get(self.pool_id).unwrap();
+	///
+	/// Worst case, iterates over [`TotalUnbondingPools`] member unbonding pools to calculate member
+	/// balance.
+	pub fn total_balance(&self) -> BalanceOf<T> {
+		let maybe_pool = BondedPool::<T>::get(self.pool_id);
+		// this internal function is always called with a valid pool id.
+		if maybe_pool.is_none() {
+			defensive!("pool should exist; qed");
+			return Zero::zero();
+		}
+		let pool = maybe_pool.expect("checked pool is not none; qed");
```

Comment:
Cleaner than using `expect` IMO.
```suggestion
		let pool = match BondedPool::<T>::get(self.pool_id) {
			Some(p) => p,
			None => {
				// this internal function is always called with a valid pool id.
				defensive!("pool should exist; qed");
				return Zero::zero();
			}
		};
```